Silicon/ApollolakePkg/Library/SocInfoLib/SocInfoLib.c

```diff
@@ -60,7 +60,7 @@ GetCpuMaxNbFrequency (
 {
   UINT64 MsrValue;
   MsrValue = AsmReadMsr64 (MSR_PLATFORM_INFO);
-  return (100 * ((MsrValue >> 8) & 0xff));
+  return (100 * (((UINT32)MsrValue >> 8) & 0xff));

```

Comment:
this is still using >>?

Reply:
UINT32  shift  is fine.  Only  UINT64 will cause issue.